nifi-standard-bundle/nifi-standard-processors/src/main/java/com/asymmetrik/nifi/processors/FetchFileSplits.java

```diff
@@ -0,0 +1,389 @@
+package com.asymmetrik.nifi.processors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.logging.LogLevel;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.OutputStreamCallback;
+import org.apache.nifi.processor.io.StreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.CopyOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+
+/**
+ * Note: this is adapted from Apache NiFi's FetchFile processor, which gets credit for handling
+ * all file processing and completion actions. This processor is primarily concerned with the
+ * adaptation of reading file splits directly as a means of handling large files more efficiently
+ * than reading them into NiFi and splitting within NiFi in a separate process.
+ */
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
```

Comment:
You could use a few more annotations, like SupportsBatching, SideEffectsFree

Reply:
Does SideEffectsFree apply here? Since we're relying on having a file on disk and we have the option to delete or move that file, I left that one off. Maybe that's the wrong assumption?